addons/mission/functions/fnc_log.sqf

```diff
@@ -0,0 +1,32 @@
+#include "script_component.hpp"
+/*
+ * Author: Mike
+ * Function based logging for 3DEN / Server
+ * Call from Functions
+ *
+ * Shows in-game warning and logs to RPT, optional logging to server/client RPT
+ *
+ * Arguments:
+ * 0: Function Name <STRING>
+ * 1: Error Message <STRING>
+ * 2: Log to Server <BOOL> (default: false)
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * ["[TAC Mission X]:", format ["Undefined value %1", _value]] call FUNC(log)
+ */
+
+params ["_functionName", "_message", ["_logToRPT", false]];
+
+private _message = [_functionName, _message] joinString " ";
+
+if (is3DENPreview) then {
+    _message call CBA_fnc_notify;
+    diag_log _message;
+} else {
+    if (_logToRPT) then {
```

Comment:
Why have this at all? If it's a live mission, we won't see any warnings then, that's bad!

Reply:
Because on a live mission, there should be no errors. Mission reviews should catch them, and if they don't we'll see in server RPT

Reply:
But we won't see in server RPT because:
- Of this condition...
- Because not every function runs on the server.

Reviews will not catch all, and I want to be able to yell at people who didn't check their missions properly.

Reply:
The condition is false for local functions, Reinforcements and Reaction would both log to server RPT. (Or everyones if someone lets it run globally)

Reply:
And again, why do we need that condition at all? It should always log to RPT.